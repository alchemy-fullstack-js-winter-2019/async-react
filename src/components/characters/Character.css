.card {
  height: 175px;
  text-shadow: 0 1px 3px rgba(0,0,0,0.6);
  color: white;
  position: relative;
  margin: 20px;

  img {
    position: absolute;
    top: 0px;
    z-index: -999;
    box-shadow: 0 8px 6px -6px black; 
  }
}