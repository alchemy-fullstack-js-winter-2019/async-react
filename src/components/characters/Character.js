import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

function Character({ character }) {
  const { name, status, species, gender, image } = character;
  
  return (
    <>
      <div className={styles.card}>
        <h2>{name}</h2>
        <p>{species}</p>
        <p>{gender}</p>
        <p>{status}</p>
        <img src={image}></img>
      </div>
    </>
  );

}

Character.propTypes = {
  character: PropTypes.object.isRequired
};


export default Character;
