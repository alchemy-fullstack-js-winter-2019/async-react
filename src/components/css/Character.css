.char {
  background-color: grey;
  text-align: center;
  margin: 5px;
}



