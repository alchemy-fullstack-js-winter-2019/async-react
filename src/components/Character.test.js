import React from 'react';
import renderer from 'react-test-renderer';
import Character from './characters/Character';

describe('Character', () => {
  it('matches a snapshot', () => {
    const character = {
      name: 'Rick',
      status: 'Alive',
      species: 'Human',
      gender: 'Male'
    };
    const tree = renderer.create(
      <Character character={character}/>
    ).toJSON();
  
    expect(tree).toMatchSnapshot();

  });
});
