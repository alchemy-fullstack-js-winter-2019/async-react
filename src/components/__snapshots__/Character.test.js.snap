// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character matches a snapshot 1`] = `
<div
  className="card"
>
  <h2>
    Rick
  </h2>
  <p>
    Human
  </p>
  <p>
    Male
  </p>
  <p>
    Alive
  </p>
  <img />
</div>
`;
